File: Talleres RESFTUL JAVA DOTNET/Dotnet/API-DOTNET-BANK/Application/Utils/PasswordHasher.cs
using System.Security.Cryptography;
using System.Text;

namespace API_DOTNET_BANK.Application.Utils
{
    public static class PasswordHasher
    {
        public static string HashPassword(string plainText)
        {
            using var sha256 = SHA256.Create();
            var bytes = Encoding.UTF8.GetBytes(plainText);
            var hash = sha256.ComputeHash(bytes);
            return Convert.ToBase64String(hash);
        }

        public static bool VerifyPassword(string plainText, string hashedPassword)
        {
            var hash = HashPassword(plainText);
            return hash == hashedPassword;
        }
    }
}

File: Talleres RESFTUL JAVA DOTNET/API_GOOGLE_MAPS_REST_JAVA_GR-03/04. SERVIDOR/api-java-bank/src/main/java/com/allpasoft/apijavabank/application/util/PasswordHasher.java
package com.allpasoft.apijavabank.application.util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public class PasswordHasher {
    
    public static String hashPassword(String plainText) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(plainText.getBytes());
            return Base64.getEncoder().encodeToString(hash);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("Error hashing password", e);
        }
    }
    
    public static boolean verifyPassword(String plainText, String hashedPassword) {
        String hash = hashPassword(plainText);
        return hash.equals(hashedPassword);
    }
}
